zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -152,9 +155,9 @@ class ZiplineLoader internal constructor(
     applicationName: String,
     manifestUrlFlow: Flow<String>,
     serializersModule: SerializersModule = EmptySerializersModule(),
-    initializer: (Zipline) -> Unit,
-  ): Flow<LoadedZipline> {
-    return flow {
+    initializer: (Zipline) -> Unit = {},
+  ): Flow<LoadResult> {
+    return channelFlow {
```

Comment:
To support throwing on failure for the event listener to catch, I needed to change this to a `channelFlow`. Are there any downstream consequences that would make this ill advised?